docs-src/docs/guides/walkthrough.md

```diff
@@ -0,0 +1,317 @@
+# An Iterative Walkthrough
+
+This guide aims to take you through three increasingly-complex deployment examples using Layer0. Successive sections build upon the previous ones, and each deployment can be completed either through using the Layer0 CLI directly, or through Terraform using our custom Layer0 Terraform Provider.
+
+We assume that you're using Layer0 v0.9.0 or later. If you have not already installed and configured Layer0, see the [installation guide](/setup/install). If you are running an older version of Layer0, you may need to [upgrade](/setup/upgrade#upgrading-older-versions-of-layer0).
+
+If you intend to deploy services using the Layer0 Terraform Provider, you'll want to make sure that you've [installed](/reference/terraform-plugin/#install) the provider correctly.
+
+**Table of Contents**:
+
+- [Deployment 1](#deployment-1-a-simple-guestbook-app): A Simple Guestbook App
+- [Deployment 2](#deployment-2-guestbook-redis): Guestbook + Redis
+- [Deployment 3](#deployment-3-guestbook-redis-consul): Guestbook + Redis + Consul
+
+
+## Deployment 1: A Simple Guestbook App
+
+In this section you'll learn how different Layer0 commands work together to deploy web applications to the cloud. The example application in this section is a guestbook -- a web application that acts as a simple message board. You can choose to complete this section using either the [Layer0 CLI](#1a-deploy-with-layer0-cli) or [Terraform](#1b-deploy-with-terraform).
+
+
+## 1a: Deploy with Layer0 CLI
+
+You will need to download the [Guestbook Task Definition](https://github.com/quintilesims/layer0-examples/blob/guides-overhaul/iterative-walkthrough/guestbook/Guestbook.Dockerrun.aws.json) file. Save that file as **Guestbook.Dockerrun.aws.json**. This walkthrough will assume that the file is located in your current working directory; if you choose to place the file elsewhere, you will need to provide the path to the file instead of just the filename when we reference it later in the guide.
+
+---
+
+### Part 1: Create the Environment
+
+The first step in deploying an application with Layer0 is to create an environment. An environment is a dedicated space in which one or more services can reside. Here, we'll create a new environment named **demo-env**. At the command prompt, execute the following:
+
+`l0 environment create demo-env`
+
+This will give us the following output:
+
+```
+ENVIRONMENT ID  ENVIRONMENT NAME  CLUSTER COUNT  INSTANCE SIZE
+demo00e6aa9     demo-env          0              m3.medium
+```
+
```

Comment:
[nit] I'm not sure if this is the best place for it - but given this is the introduction to layer0, maybe this is where we would show them `environment get` and `environment list`? So we can be like "hey, if you want to see your environments, use `environment list`. notice the `api` environment is there. If you want detailed info, use `environment get`, blah blah" - and we do a similar thing for each entity we create throughout this intro section.